[docs] Add more instructions to third party dependency docs (#25208)
expo/expo: docs/pages/modules/third-party-library.mdx

description: Learn how to create a simple wrapper around two separate native libraries using Expo Modules.
---

> import { Collapsible } from '~/ui/components/Collapsible';

Reviewer: ```suggestion
import { CODE } from '~/ui/components/Text';
import { Collapsible } from '~/ui/components/Collapsible';
```

    # Swift/Objective-C compatibility
```

> <Collapsible summary="Are you trying to use a .framework dependency?">

Reviewer: ```suggestion
<Collapsible summary={<>Are you trying to use a <CODE>.framework</CODE> dependency?</>}>
```

Let's use inline syntax highlight here. I've also added the import statement above.

> On iOS you can also use dependencies bundled as a framework by using the `vendored_framework` config option.

Reviewer: ```suggestion
On iOS, you can also use dependencies bundled as a framework by using the `vendored_framework` config option.
```

Missing comma here.

```diff ios/ExpoRadialChart.podspec
    s.static_framework = true
    s.dependency 'ExpoModulesCore'
+   s.vendored_frameworks 'Frameworks/MyFramework.framework'
    # Swift/Objective-C compatibility
```

</Collapsible>

> <Collapsible summary="Are you trying to use a .aar dependency?">

Reviewer: ```suggestion
<Collapsible summary={<>Are you trying to use a <CODE>.aar</CODE> dependency?</>}>
```

Similarly here we can also use syntax highlight.

> Create a `libs` folder in the `android` folder of your Expo module. Place your `aar` file there.
> 
> Next add the folder as a repository like so:

Reviewer: ```suggestion
Inside the **android** directory, create another directory called **libs** and place the **.aar** file inside it. Then, add the folder as a repository:
```

Let's be concise here.

```diff android/build.grade
  repositories {
    mavenCentral()
+   flatDir {
+       dirs 'libs'
+   }
  }
```

> Last, add the dependency to the `dependencies` list (using a package path, not a filename, but note the `@aar` at the end):

Reviewer: ```suggestion
Finally, add the dependency to the `dependencies` list. Instead of the filename, use the package path, which includes the `@aar` at the end:
```

Improved verbiage here as to what final step to take, and also the info inside the brackets should be part of the main flow so that a developer doesn't ignore it.

    # Swift/Objective-C compatibility
```

> <Collapsible summary={<>Are you trying to use a <CODE>.framework</CODE> dependency?</>}>

Author: ```suggestion
<Collapsible summary={<>Are you trying to use an <CODE>.xcframework</CODE> or <CODE>.framework</CODE> dependency?</>}>
```
